### ISSUE TITLE
Proposal: Podcast and PodcastEpisode

### ISSUE THREAD

**Issue by Racum** [schemaorg/schemaorg#373]
## Introduction

New schema types: **Podcast** and **PodcastEpisode**.

Podcasts are part of the common Internet slang for more than 10 years, they are everywhere, they are slowly becoming very popular, but they are not quite well represented by the Semantic Web.

So far, if anyone wanted to classify a podcast page, it suppose to use a generic “CreativeWork” type for the list of content and an “Episode” type for each instance. Expanding CreativeWork into “Podcast” type gives it the proper meaning, plus a few useful properties, and, specifying Episode into “PodcastEpisode” makes clear the difference between it and other episodes (RadioEpisode/TVEpisode).
### Definition:

A podcast, referring to a main theme containing a series of related sequential episodes.
### Inheritance

`Thing > CreativeWork > Podcast`
### Properties

| Property | Expected Type | Description |
| --- | --- | --- |
| `network` | Organization | An Organization specialized in provide or host a group of Podcasts, not only this one. “Directories” (like iTunes) does not qualify. |
| `category` | Text | The category of the Podcast. It is a good practice to follow the [taxonomy from iTunes](https://www.apple.com/itunes/podcasts/specs.html#category). |
| `rssFeed` | URL | RSS Feed URL. |
### Mapping from RSS "channel" sub-tags:

Just for reference:

| RSS Tag | Property | from |
| --- | --- | --- |
| `<title>` | name | Thing |
| `<description>` | description | Thing |
| `<link>` | url | Thing |
| `<language>` | inLanguage | CreativeWork |
| `<copyright>` | copyrightHolder | CreativeWork |
| `<managingEditor>` | editor | CreativeWork |
| `<lastBuildDate>` | dateModified | CreativeWork |
| `<image>` | image | Thing |
| `<category>` | category | Podcast |
| `<itunes:author>` | author | CreativeWork |
| `<itunes:category>` | category | Podcast |
| `<itunes:image>` | image | Thing |
| `<itunes:explicit>` | [not] isFamilyFriendly | CreativeWork |
| `<itunes:owner>` | accountablePerson | CreativeWork |
| `<itunes:subtitle>` | alternativeHeadline | CreativeWork |
| `<itunes:summary>` | text | CreativeWork |
| N/A | rssFeed | Podcast |
| N/A | network | Podcast |
### Notes
- The following `<channel>` sub-tags aren't mapped because they are implementation details:
  - `<webMaster>`
  - `<generator>`
  - `<docs>`
  - `<ttl>`
  - `<itunes:block>`
  - `<itunes:complete>`
  - `<itunes:new-feed-url>`
### Definition:

A podcast episode, referring to a single instance of multimedia content (audio or video).
### Inheritance

`Thing > CreativeWork > Episode > PodcastEpisode`
### Properties

| Property | Expected Type | Description |
| --- | --- | --- |
| `podcast` | Podcast | The Podcast that contains this PodcastEpisode. |
| `guid` | Text | Permanent, case-sensitive "Globally Unique Identifier". |
| `category` | Text | The category of the PodcastEpisode (in case it differs from Podcast's category). It is a good practice to follow the [taxonomy from iTunes](https://www.apple.com/itunes/podcasts/specs.html#category). |
### Mapping from RSS "item" sub-tags:

Just for reference:

| RSS Tag | Property | from |
| --- | --- | --- |
| `<title>` | name | Thing |
| `<description>` | description | Thing |
| `<link>` | url | Thing |
| `<guid>` | guid | PodcastEpisode |
| `<author>` | author | CreativeWork |
| `<pubDate>` | datePublished | CreativeWork |
| `<comments>` | discussionUrl | CreativeWork |
| `<enclosure>` | associatedMedia | CreativeWork |
| `<category>` | category | PodcastEpisode |
| `<itunes:author>` | author | CreativeWork |
| `<itunes:image>` | image | Thing |
| `<itunes:duration>` | timeRequired | CreativeWork |
| `<itunes:explicit>` | [not] isFamilyFriendly | CreativeWork |
| `<itunes:order>` | episodeNumber | Episode |
| `<itunes:subtitle>` | alternativeHeadline | CreativeWork |
| `<itunes:summary>` | text | CreativeWork |
| N/A | podcast | PodcastEpisode |
### Notes
- The following `<item>` sub-tags aren't mapped because they are implementation details:
  - `<itunes:block>`
  - `<itunes:isClosedCaptioned>`
## Example

Adapted from [http://www.relay.fm/inquisitive/27](http://www.relay.fm/inquisitive/27).
### Without Markup

``` html
<div class="show__description">
  <img src="https://.../inquisitive_artwork.png">
  <h2><a href="http://www.relay.fm/inquisitive"><span>Inquisitive</span></a></h2>
  <h3>
    <p>
      Inquisitive is a show for the naturally curious.
      Each week, Myke Hurley takes a look at what makes creative
      people successful and what steps they have taken to get there.
    </p>
  </h3>
  <h5>Subscribe</h5>
  <ul>
    <li><a href="http://www.relay.fm//inquisitive/feed">RSS</a></li>
    <li><a href="https://itunes.apple.com/us/podcast/id909109678">iTunes</a></li>
  </ul>
  <h5>Hosted By<spam>Myke Hurley</span></h5>
</div>

<div class="episode__entry">
  <h2>
    <a href="http://www.relay.fm/inquisitive/27">
      <span>#27: Behind the App #1: History</span>
    </a>
  </h2>
  <small>February 18th, 2015</small>
  <small>37 minutes</small>
  <p>
    In the first episode of “Behind the App”, a special series of
    Inquisitive, we take a look at the beginnings of iOS app development,
    by focusing on the introduction of the iPhone and the App Store.
  </p>
  <p>
    Download:
    <a href="http://.../Inquisitive_027.mp3">MP3 (25.75 MB)</a>
  </p>
</div>
```
### Microdata

``` html
<div itemscope itemtype="http://schema.org/Podcast" class="show__description">
  <img itemprop="image" src="https://.../inquisitive_artwork.png">
  <h2>
    <a itemprop="url" href="http://www.relay.fm/inquisitive">
      <span itemprop="name">Inquisitive</span>
    </a>
  </h2>
  <h3>
    <p itemprop="description">
      Inquisitive is a show for the naturally curious.
      Each week, Myke Hurley takes a look at what makes creative
      people successful and what steps they have taken to get there.
    </p>
  </h3>
  <h5>Subscribe</h5>
  <ul>
    <li><a itemprop="rssFeed"
        href="http://www.relay.fm//inquisitive/feed">RSS</a></li>
    <li><a href="https://itunes.apple.com/us/podcast/id909109678">iTunes</a></li>
  </ul>
  <h5>Hosted By<spam itemprop="author" >Myke Hurley</span></h5>
</div>

<div itemscope itemtype="http://schema.org/PodcastEpisode"  class="episode__entry">
  <h2>
    <a itemprop="url" href="http://www.relay.fm/inquisitive/27">
      <span itemprop="name">#27: Behind the App #1: History</span>
    </a>
  </h2>
  <small itemprop="datePublished" datetime="2015-02-18">February 18th, 2015</small>
  <small itemprop="duration" datetime="PT27M">37 minutes</small>
  <p itemprop="description">
    In the first episode of “Behind the App”, a special series of
    Inquisitive, we take a look at the beginnings of iOS app development,
    by focusing on the introduction of the iPhone and the App Store.
  </p>
  <p itemprop="associatedMedia" itemscope itemtype="http://schema.org/MediaObject">
    Download:
    <a itemprop="contentUrl" href="http://.../Inquisitive_027.mp3">
      MP3 (<span itemprop="contentSize">25.75 MB</span>)
    </a>
  </p>
</div>
```

**Comment by chaals** [comment]
I'm not convinced by the `category` property. Being able to link to a popular online store is handy, but baking a particular one into schema.org doesn't feel right - there are a lot of subject classification schemes, and apart from anything else I don't find Apple's terribly good, so I would suggest using `about` for now (while letting people add the itunes category or their own equivalent, if they are using RDFa or JSON-LD to mix vocabs.

Same for `network` - I am not sure what it does beyond `publisher`.

And for `guid` I am just sceptical we need it. I note it isn't in the examples, and I am not sure what we would use as a best practice value. I suspect we can rely on existing identification instead.

Beyond that, the proposal itself seems rational. The other approach would be to explicitly state that we think podcasts are the same as radio (which is how I think of them - I also listen to as much radio streamed over the net as I do AM or FM broadcasts of audio signals), and just say "use the radio properties".

Thoughts?

**Comment by vholland** [comment]
I agree with @chaals re: network and category. CreativeWork already has a "genre" property. You could use that.

If you change your hierarchy a bit to Thing > CreativeWork > Series > Podcast, you could use the existing properties to link up the Episodes to the Series, so you would not need the "podcast" property.

**Comment by jvandriel** [comment]
In regards to:
| Property | Expected Type | Description |
|----------|---------------|-------------|
| guid | Text | Permanent, case-sensitive "Globally Unique Identifier". |

Wouldn't that be the same as using @itemid, @resource or @id?

**Comment by Dataliberate** [comment]
On 20 Mar 2015, at 11:59, chaals <notifications@github.com<mailto:notifications@github.com>> wrote:

I'm not convinced by the category property. Being able to link to a popular online store is handy, but baking a particular one into schema.org<http://schema.org> doesn't feel right - there are a lot of subject classification schemes, and apart from anything else I don't find Apple's terribly good, so I would suggest using about for now

Agree  - don’t see the special case here that is made just for Podcasts.  Using about to reference a value in a classification scheme(s) seems to be OK for most other CreativeWorks

(while letting people add the itunes category or their own equivalent, if they are using RDFa or JSON-LD to mix vocabs.

Same for network - I am not sure what it does beyond publisher.

Neither do I - its just the publishing organisation.

And for guid I am just sceptical we need it. I note it isn't in the examples, and I am not sure what we would use as a best practice value. I suspect we can rely on existing identification instead.

Yes surely the URI of the of the description is the globally unique ID.


Beyond that, the proposal itself seems rational. The other approach would be to explicitly state that we think podcasts are the same as radio (which is how I think of them - I also listen to as much radio streamed over the net as I do AM or FM broadcasts of audio signals), and just say "use the radio properties".

I have some sympathy with this approach - it feels a bit like needing an AudioSeries Type to be a super type for RadioSeries and a new PodcastSeries - then Episode would work for both.

~Richard

Thoughts?

**Comment by Racum** [comment]
@chaals

About ```category```: I agree about not recommend the Apple’s list, but lets keep it a open field and assume the Apple-based ones as legal values, since it is a “de-facto” standard in the podcast world.

About ```network```: it IS a publisher, I didn’t notice when I was writing the proposal, I agree about omit this field.


@chaals / @jvandriel / @Dataliberate

About ```guid```: must podcasts just use the episode page URL as am identifier, in this case it is the same as the ```url``` field. Some documentation recommend the use of UUIDs, but nobody actually uses it, I could not find an example different from ```url```. I’m ok about omitting it as well


@vholland / @Dataliberate

The only extra fields Series give to CreativeWork are ```startDate```and ```endDate```,  and podcasts usually don’t have these concepts, there are cases of podcasts paused by years that just restart. Also the other children of Series (BookSeries, MovieSeries…) don’t behave like a Podcast. I think it makes more sense to keep a child of CreativeWork.


TL:DR:

* No formal Apple recommendation for category, but keep as a open field;
* Out with Podcast.network, prefer publisher;
* Out with Episode.guid, url does the trick;
* Keep Podcast as a child of CreativeWork.

Thoughts?

**Comment by chaals** [comment]
I think, as Richard suggested, that `about` would work instead of `category` - but there is no reason not to use the apple scheme in that, and it would make a reasonable example.

**Comment by rvguha** [comment]
We should collect some data.

If indeed podcasts are described using a certain categorization scheme,
given that describing podcasts is right in the middle of our charter, we
should create a relation for that categorization scheme.

guha

**Comment by Racum** [comment]
@chaals / @rvguha

**Personally**, I don't care too much about what categories to use, I'm way more concern about simply **have** the Podcast/PodcastEpisode schemes. But, putting personal opinions aside, I made some research about the categories, I hope it helps the discussion:

The podcast sites themselves usually don’t use categorization other than the internal categories of the site or blog, but podcast directories tend to follow the Apple convention, sometimes even down to the sub-categories, a quick googling got me these examples:

* podcastdirectory.com
    * [List of categories and sub-categories](http://www.podcastdirectory.com/)
    * [Podcast page: category breadcrumb on the left sidebar](http://www.podcastdirectory.com/podcasts/microbeworld-video-27859.html)
* learnoutloud.com
    * [List of categories](http://www.learnoutloud.com/Podcast-Directory)
    * [Podcast page: category breadcrumb on the top](http://www.learnoutloud.com/Podcast-Directory/History/-/Serial-Podcast/78735)
* ipodder.org
    * [List of categories](http://www.ipodder.org/directory/4/podcasts)
    * [Podcast page: category breadcrumb on the top](http://www.ipodder.org/p/A-Buddhist-Podcast)

For reference, [Apple's list is here](https://www.apple.com/itunes/podcasts/specs.html) (at the end of the page).

I can say that Apple's categorization "it's a thing"! Apple was the first to try to put some other in the podcast chaos about 10 years ago; their patterns stuck and were copied by many other directories after that.

**Comment by dhilowitz** [comment]
Any progress on this?

**Comment by Racum** [comment]
@dhilowitz, the maintainers of the project commented on my initial proposal, I am waiting on their opinion about the Apple's category list (see above), but they are deliberating on this for months... I think I could just try my luck with a pull request!

**Comment by rvguha** [comment]
Just as an FYI, all deliberations, if any, take place on this list. So, if
you don't see any, there hasn't been any.

guha

**Comment by unor** [comment]
I think `rssFeed` should be renamed to `feed` (or `webFeed`, if `feed` would be too ambiguous); making it technology-neutral (ref. [Atom](https://en.wikipedia.org/wiki/Atom_%28standard%29)).

**Comment by Racum** [comment]
@unor, I agree! ...just `feed` is more elegant and doesn't look to collide with any other schema.

**Comment by rvguha** [comment]
feed is too ambiguous.

I prefer rssFeed, but may be biased.

guha

**Comment by Racum** [comment]
@unor, @rvguha: in podcast lingo, "feed" and "rss" are used interchangeably. Both `rssFeed` and `feed` make sense, it just a matter of choose between implicit vs explicit; is there any similar case solved in other schemas to use as reference?

**Comment by chaals** [comment]
Agree that feed is way too ambiguous for schema.

Why doesn't isPartOf cover this use case?
(sleepy, so the answer might be obvious…)

**Comment by Racum** [comment]
@chaals 

From the definition: "isPartOf" - Indicates a CreativeWork that this CreativeWork is (in some sense) part of.

https://schema.org/isPartOf

I don't thing the feed itself is another "CreativeWork", it is just an already machine-readable alternative representation.

**Comment by chaals** [comment]
I'm not sure isPartOf is right, but I don't think "feed" is either. Not just because the name is too generic for the concept we are using here as specific to podcasts - the concept is more general... Perhaps we shuld look at the stuff for serial publication including TV series etc.

A feed *is* a creative work, in the same way an exhibition, blog (as distinct from blogpost) or an album is.

Schema should not get all FRBR in describing the intricate relationships between different manifestations of works and collections (inter alia, because FRBR already does it ;) ). But we should handle basics in a rational way instead of having parallel silos, if we can.

**Comment by vholland** [comment]
+1 for rssFeed to future proof against other types of feeds.

Why do we need guid? It seems "@id" or itemid as appropriate in the markup?

If Podcast is a subtype of http://schema.org/CreativeWorkSeries, we already have the link from Podcast to PodcastEpisode without introducing new properties. We could also reuse http://schema.org/publisher to avoid adding "network".

**Comment by Racum** [comment]
Putting all the comments together, I think we got a simpler specification:

* No longer `Podcast.network` (use CreativeWork.publisher instead).
* No longer `PodcastEpisode.guid` (omitted).
* Attribute `category` is open-ended, there is no formal recommendation to use Apple's taxonomy, although probably the podcasts tend to use it anyway.

What do you all think?

## Podcast:

`Thing > CreativeWork > Podcast`

* `rssFeed`: URL
* `category`: Text

## PodcastEpisode

`Thing > CreativeWork > Episode > PodcastEpisode`

* `podcast`: Podcast
* `category`: Text

**Comment by srimola** [comment]
+1

**Comment by collinwu** [comment]
+1

**Comment by iluxan** [comment]
What are the next steps here? Is this proposal still planned or abandoned?

**Comment by Racum** [comment]
I didn't find a contributing guide, I don't know if just need to send a PR or do anything else before.

**Comment by HipsterJazzbo** [comment]
Hi, I'm pretty interested in this. Is there some procedure to move this on? I'm happy to do some legwork if someone can point me in the right direction.

**Comment by Racum** [comment]
It is probably just a matter of editing the `/data/schema.rdfa` file, but I'm not familiar with this format, so I have no way properly test it before create a PR.

**Comment by HipsterJazzbo** [comment]
Is there like, no one who runs this repo who can answer this question?

**Comment by RichardWallis** [comment]
In principle - submit a Pull Request, of vocab changes accompanied by examples, for consideration in an upcoming release.  You may want to do this in files named issue-373.rdfa and issue-examples.txt instead of submitting edits to the main schema.rdfa and examples.txt. This also helps if the proposal is initially placed in the [pending area](http://pending.schema.org/).

In practice - you may find the following "schema.org in practice" articles: [part 1](http://dataliberate.com/2016/02/10/evolving-schema-org-in-practice-pt1-the-bits-and-pieces/), [2](http://dataliberate.com/2016/02/25/evolving-schema-org-in-practice-pt2-working-within-the-vocabulary/), and [3](http://dataliberate.com/2016/03/01/evolving-schema-org-in-practice-pt3-choosing-where-to-extend/), a help in how you can try/test things locally, and the [HowWeWork](http://schema.org/docs/howwework.html) document for general background. 

~Richard

**Comment by Racum** [comment]
@RichardWallis you just wrote a perfect draft for a `CONTRIBUTING.md` file! :)

**Comment by UmamiDearest** [comment]
@Racum Echoing all those expressing more and more interest in this becoming part of the schema landscape. Any update on your submission @Racum?

**Comment by Aaranged** [comment]
@vholland Did you mean something like `PostcastSeries` for that `CreativeWorkSeries` subclass, or are you thinking that `Postcast` itself is a `CreativeWorkSeries`?  And if the latter, what would the type be for an individual podcast?

**Comment by vholland** [comment]
I was thinking the type schema.org/Podcast would be a subclass of schema.org/CreativeWorkSeries. This keeps it consistent so _This American Life_ is a series whether you are discussing the radio series or the modern podcast version.

An individual "podcast" would be a PodcastEpisode as described above. As an example _This American Life 625: Essay B_ is an episode.

I am aware that some podcast series will have only a single episode.

**Comment by RichardWallis** [comment]
To avoid confusion  would they not be better as PodcastSeries and PodcastEpisode?

**Comment by vholland** [comment]
@RichardWallis Those names sound good to me

**Comment by dougblackjr** [comment]
Has there been any movement on this?

**Comment by JeffreyBurns** [comment]
+1 for including PodcastSeries and PodcastEpisode

**Comment by machito** [comment]
+1

**Comment by baq** [comment]
What's the status on this?  +1

**Comment by vholland** [comment]
I was talking to someone about podcasts and schema.org and was embarrassed to realize we have been discussing for 3 years. In an effort to move this along, I created PR #2007

**Comment by mariotoledo** [comment]
While this is still an open issue, which is the best option for markup my podcast website? I'm thinking of using CreativeSeries. Does anyone has a better idea?

**Comment by inetbiz** [comment]
> @vholland Did you mean something like `PostcastSeries` for that `CreativeWorkSeries` subclass, or are you thinking that `Postcast` itself is a `CreativeWorkSeries`? And if the latter, what would the type be for an individual podcast?
@Aaranged Any news on this proposal that you know of?

**Comment by gr0g** [comment]
+1

**Comment by Abir-H** [comment]
+1

Any progress with this?

**Comment by RichardWallis** [comment]
Implemented in [Release 4.0](https://schema.org/docs/releases.html#v4.0)

**Comment by DamonHD** [comment]
Should 'trailer' not be allowed to be an AudioObject or even a WebPage or Article or any other CreativeWork also?  Especially when promoting an audio podcast, a video trailer seems overblown (and doesn't fit what I want to do right now)!

**Comment by DamonHD** [comment]
From my use of webFeed, I think that it should be allowed in CreativeWorkSeries (for a series which gets a RSS/Atom feed, but isn't a podcast) and PodcastEpisode (to link to the podcast's feed) and indeed WebPage and Article and maybe all CreativeWork items to allow them to refer to a RSS/Atom feed associated with or that has that item as an entry/episode...
